lib/grpnames.gd

```diff
@@ -122,6 +142,21 @@
 ##   and where it is in the b-sequence if any. The the
 ##   linear algorithm above may be used.
 ##
+
+## The Kayal-Nezhmetdinov algorithm is described in
+## Neeraj Kayal and Timur Nezhmetdinov, Factoring Groups Efficiently, in
+## International Colloquium on Automata, Languages and Programming (ICALP) ,
+## Springer Verlag, 2009.
+# Gives a normal complement to the second argument if exists, fail otherwise.
+# In theory it finds the normal complement for infinite groups, as well,
+# but has an infinite loop if g/N is Abelian for infinite N.
+DeclareOperation( "ComplementNormalSubgroup", [IsGroup, IsGroup]);
+DeclareOperation( "ComplementNormalSubgroupNC", [IsGroup, IsGroup]);
```

Comment:
First, I did not compare this with ComplementClassesRepresentatives. If you want, I will figure out some tests for it. 

Second, ComplementNormalSubgroups does not compute a list (ever), it only computes a normal complement if exist, and outputs fail if none exist. As this should be a normal subgroup, all conjugacy classes basically consist of only 1 element, anyway. 

As for how many complements exist and if we really want to compute all: 
If the group is NxB (B is one complement for N), then every other complement is $\{b\cdot phi(b) : b \in B\}$, where $\phi \colon B \to Z(N)$ is a homomorphism. There could be a lot of these if the center is big, and I am not sure if computing them all makes any sense. One would not necessarily need all direct decompositions, only one particular. Anyway, I can write another function outputting a list of all complements, if you think it would be useful. 

As for the name: I do not mind changing it, but I would rather keep 'normal' in the name to reflect that it finds a normal complement.